code/modules/projectiles/guns/flamer.dm

```diff
@@ -14,368 +14,302 @@
+		if(isxeno(mob_caught))
+			var/mob/living/carbon/xenomorph/xeno_caught = mob_caught
+			if(CHECK_BITFIELD(xeno_caught.xeno_caste.caste_flags, CASTE_FIRE_IMMUNE))
 				continue
-			fire_mod = X.get_fire_resist()
-		else if(ishuman(M))
-			var/mob/living/carbon/human/H = M //fixed :s
-
+			fire_mod = xeno_caught.get_fire_resist()
+		else if(ishuman(mob_caught))
+			var/mob/living/carbon/human/human_caught = mob_caught
 			if(user)
-				if(!user.mind?.bypass_ff && !H.mind?.bypass_ff && user.faction == H.faction)
-					log_combat(user, H, "flamed", src)
-					user.ff_check(30, H) // avg between 20/40 dmg
-					log_ffattack("[key_name(user)] flamed [key_name(H)] with \a [name] in [AREACOORD(T)].")
-					msg_admin_ff("[ADMIN_TPMONTY(user)] flamed [ADMIN_TPMONTY(H)] with \a [name] in [ADMIN_VERBOSEJMP(T)].")
+				if(!user.mind?.bypass_ff && !human_caught.mind?.bypass_ff && user.faction == human_caught.faction)
+					log_combat(user, human_caught, "flamed", src)
+					user.ff_check(30, human_caught) // avg between 20/40 dmg
+					log_ffattack("[key_name(user)] flamed [key_name(human_caught)] with \a [name] in [AREACOORD(turf_to_ignite)].")
+					msg_admin_ff("[ADMIN_TPMONTY(user)] flamed [ADMIN_TPMONTY(human_caught)] with \a [name] in [ADMIN_VERBOSEJMP(turf_to_ignite)].")
 				else
-					log_combat(user, H, "flamed", src)
+					log_combat(user, human_caught, "flamed", src)
 
-			if(H.hard_armor.getRating("fire") >= 100)
+			if(human_caught.hard_armor.getRating("fire") >= 100)
 				continue
 
-		M.take_overall_damage_armored(rand(burn,(burn*2))* fire_mod, BURN, "fire", updating_health = TRUE) // Make it so its the amount of heat or twice it for the initial blast.
-		M.adjust_fire_stacks(rand(5,burn*2))
-		M.IgniteMob()
+		mob_caught.take_overall_damage_armored(rand(burn_level, (burn_level * mob_flame_damage_mod)) * fire_mod, BURN, "fire", updating_health = TRUE) // Make it so its the amount of heat or twice it for the initial blast.
+		mob_caught.adjust_fire_stacks(rand(5, (burn_level * mob_flame_damage_mod)))
+		mob_caught.IgniteMob()
 
-		to_chat(M, "[isxeno(M)?"<span class='xenodanger'>":"<span class='highdanger'>"]Augh! You are roasted by the flames!")
+		var/burn_message = "Augh! You are roasted by the flames!"
+		to_chat(mob_caught, isxeno(mob_caught) ? span_xenodanger(burn_message) : span_highdanger(burn_message))
 
-/obj/item/weapon/gun/flamer/get_ammo_type()
-	if(!ammo)
-		return list("unknown", "unknown")
-	else
-		return list(ammo.hud_state, ammo.hud_state_empty)
+/obj/item/weapon/gun/flamer/big_flamer
+	name = "\improper M240A1 incinerator unit"
+	desc = "The M240A1 has proven to be one of the most effective weapons at clearing out soft-targets. This is a weapon to be feared and respected as it is quite deadly."
+	icon_state = "m240"
+	item_state = "m240"
 
-/obj/item/weapon/gun/flamer/get_ammo_count()
-	if(!current_mag)
-		return 0
-	else
-		return current_mag.current_rounds
+/obj/item/weapon/gun/flamer/mini_flamer
```

Comment:
you have to nerf the mini_flamer, no?

Reply:
nope. I'll leave that to mjp or someone smarter than me